src/internal/operations/backup.go

```diff
@@ -632,23 +633,98 @@ func getNewPathRefs(
 	return newPath, newLoc, updated, nil
 }
 
-func mergeDetails(
+// mergeDetailsFromAssistBackups merges the deets from the assist backups into
+// the current deets.
+func mergeDetailsFromAssistBackups(
 	ctx context.Context,
 	detailsStore streamstore.Streamer,
-	backups []kopia.BackupEntry,
-	dataFromBackup kopia.DetailsMergeInfoer,
 	deets *details.Builder,
-	writeStats *kopia.BackupStats,
+	assistBackups []kopia.BackupEntry,
 	serviceType path.ServiceType,
 	errs *fault.Bus,
-) error {
-	detailsModel := deets.Details().DetailsModel
+) (map[string]details.Entry, error) {
+	// If the backup operation did not produce any new details, or if we don't
+	// have any assist backups, we don't need to merge anything.
+	if deets.Empty() || len(assistBackups) == 0 {
+		logger.Ctx(ctx).Infow("no assist backups to merge",
+			"deets_empty", deets.Empty(),
+			"num_assist_backups", len(assistBackups))
+
+		return nil, nil
+	}
 
-	// getting the values in writeStats before anything else so that we don't get a return from
-	// conditions like no backup data.
-	writeStats.TotalNonMetaFileCount = len(detailsModel.FilterMetaFiles().Items())
-	writeStats.TotalNonMetaUploadedBytes = detailsModel.SumNonMetaFileSizes()
+	repoRefToAssistDeets := make(map[string]details.Entry)
+
+	for _, assistBackup := range assistBackups {
+		mctx := clues.Add(ctx, "assist_backup_id", assistBackup.ID)
 
+		baseDeets, err := getDetailsFromBackup(
+			mctx,
+			assistBackup.Backup,
+			detailsStore,
+			errs)
+		if err != nil {
+			return nil, clues.New("fetching details for assist backup")
+		}
+
+		for _, entry := range baseDeets.Items() {
+			rr, err := path.FromDataLayerPath(entry.RepoRef, true)
+			if err != nil {
+				return nil, clues.New("parsing base item info path").
+					WithClues(mctx).
+					With("repo_ref", path.NewElements(entry.RepoRef))
```

Comment:
nit: I think LoggableDir may be a little clearer here. It'll also properly handle trailing slashes (I guess I should go make a PR for that for `NewElements`)